starter-code/src/components/Ironcontacts.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from "react";
+
+import "./Ironcontacts.css";
+import contacts from "../contacts.json";
+
+class Ironcontacts extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { contactsArray: contacts };
+  }
+
+  addContact() {
+    const newContact = prompt("Animal Name?");
+
+    // update the array
+    const contact = this.state.contactsArray;
+    contact.push(newContact);
+
+    // setState() to tell React to changen the DOM);
+    this.setState({ contactsArray: contact });
+  }
+
+  randomContact() {
+    const randomIndex = Math.floor(Math.random() * contactsArray.length);
```

Comment:
Here you're using again the contactsArray from your constructor so you have to write it with `this.state` like this : 
```
const randomIndex = Math.floor(Math.random() * this.state.contactsArray.length);
this.addContact(this.state.contactsArray[randomIndex]);
```